### PR TITLE
tweak 4MemberFullRosca to include situation where contract balance is less than what the member can withdraw

### PR DESCRIPTION
**Pull request by shine2lay**
This change is [Reviewable](https://reviewable.io/reviews/wetrustplatform/rosca-contracts/47)

### CODE REVIEW

**Comment by rmerom** [comment]
Review status: 0 of 1 files reviewed at latest revision, 4 unresolved discussions.

---

*[test/4MemberFullRoscaTest.js, line 239 at r1](https://reviewable.io:443/reviews/wetrustplatform/rosca-contracts/47#-KZyU8yq4_wF8uf5vOWc:-KZyU8yq4_wF8uf5vOWd:b7125zp) ([raw file](https://github.com/wetrustplatform/rosca-contracts/blob/0e86db1018c6ff045a942596f55fba032f9304eb/test/4MemberFullRoscaTest.js#L239)):*
> ```JavaScript
>   }));
>   
>   it("4th round (last): nodoby bids and p3, the only non-winner, can't win as he's not in good standing",
> ```

maybe add something to the description?

---

*[test/4MemberFullRoscaTest.js, line 241 at r1](https://reviewable.io:443/reviews/wetrustplatform/rosca-contracts/47#-KZyUBEfi3YPcLJmtF4T:-KZyUBEfi3YPcLJmtF4U:b-7kia0g) ([raw file](https://github.com/wetrustplatform/rosca-contracts/blob/0e86db1018c6ff045a942596f55fba032f9304eb/test/4MemberFullRoscaTest.js#L241)):*
> ```JavaScript
>   it("4th round (last): nodoby bids and p3, the only non-winner, can't win as he's not in good standing",
>       co(function*() {
>     // 0's credit is = 6.91C
> ```

s/0/p

---

*[test/4MemberFullRoscaTest.js, line 245 at r1](https://reviewable.io:443/reviews/wetrustplatform/rosca-contracts/47#-KZyUIXZU8otQ7rgnruw:-KZyUIXZU8otQ7rgnrux:bmxd8s4) ([raw file](https://github.com/wetrustplatform/rosca-contracts/blob/0e86db1018c6ff045a942596f55fba032f9304eb/test/4MemberFullRoscaTest.js#L245)):*
> ```JavaScript
>     // newCredit + TD / MC == 3C => newCredit == 4C - TD / MC == 4C - 0.15 * 4C / 4 == 3.85C.
>     // They should be able thus to withdraw 6.91C - 3.85C == 3.06C.
>     // Since contract's balance is only 2.424, p0's credit after withdraw 6.91C - 2.424 == 4.486
> ```

plz add C to last 2 numbers

---

*[test/4MemberFullRoscaTest.js, line 302 at r1](https://reviewable.io:443/reviews/wetrustplatform/rosca-contracts/47#-KZyU_AGZbO9dm0pD0ml:-KZyU_AGZbO9dm0pD0mm:b165yi8) ([raw file](https://github.com/wetrustplatform/rosca-contracts/blob/0e86db1018c6ff045a942596f55fba032f9304eb/test/4MemberFullRoscaTest.js#L302)):*
> ```JavaScript
>     // p3: nothing (still owes 0.85C, even though has not won any round)
>     
>     // Let p0 and p1 withdraw.
> ```

plz update

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/wetrustplatform/rosca-contracts/47)*

**Review by rmerom — APPROVED** [review]
Thanks for doing this so quickly!
Plz fix comments before merging (no need for another review iteration on my side)

**Comment by shine2lay** [comment]
Review status: 0 of 1 files reviewed at latest revision, 4 unresolved discussions.

---

*[test/4MemberFullRoscaTest.js, line 239 at r1](https://reviewable.io:443/reviews/wetrustplatform/rosca-contracts/47#-KZyU8yq4_wF8uf5vOWc:-KZyVbcivqlFCR9IcIDA:b-896fix) ([raw file](https://github.com/wetrustplatform/rosca-contracts/blob/0e86db1018c6ff045a942596f55fba032f9304eb/test/4MemberFullRoscaTest.js#L239)):*
<details><summary><i>Previously, rmerom (Ron Merom) wrote…</i></summary><blockquote>

maybe add something to the description?
</blockquote></details>

Done.

---

*[test/4MemberFullRoscaTest.js, line 241 at r1](https://reviewable.io:443/reviews/wetrustplatform/rosca-contracts/47#-KZyUBEfi3YPcLJmtF4T:-KZyVNzvjbP_4v4mWgpS:b-896fix) ([raw file](https://github.com/wetrustplatform/rosca-contracts/blob/0e86db1018c6ff045a942596f55fba032f9304eb/test/4MemberFullRoscaTest.js#L241)):*
<details><summary><i>Previously, rmerom (Ron Merom) wrote…</i></summary><blockquote>

s/0/p
</blockquote></details>

Done.

---

*[test/4MemberFullRoscaTest.js, line 245 at r1](https://reviewable.io:443/reviews/wetrustplatform/rosca-contracts/47#-KZyUIXZU8otQ7rgnruw:-KZyVNfN7RKdpY2q325P:b-896fix) ([raw file](https://github.com/wetrustplatform/rosca-contracts/blob/0e86db1018c6ff045a942596f55fba032f9304eb/test/4MemberFullRoscaTest.js#L245)):*
<details><summary><i>Previously, rmerom (Ron Merom) wrote…</i></summary><blockquote>

plz add C to last 2 numbers
</blockquote></details>

Done.

---

*[test/4MemberFullRoscaTest.js, line 302 at r1](https://reviewable.io:443/reviews/wetrustplatform/rosca-contracts/47#-KZyU_AGZbO9dm0pD0ml:-KZyViOmPmRKojRSLDIv:b-896fix) ([raw file](https://github.com/wetrustplatform/rosca-contracts/blob/0e86db1018c6ff045a942596f55fba032f9304eb/test/4MemberFullRoscaTest.js#L302)):*
<details><summary><i>Previously, rmerom (Ron Merom) wrote…</i></summary><blockquote>

plz update
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/wetrustplatform/rosca-contracts/47)*

**Pull request merged**
